### PR TITLE
3764 update message broker overview page

### PR DESCRIPTION
**Pull request by EugeneTseitlin**
Closes: #3764 

## PR Details

Added tiles and request for connected services

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Topics/TopicList.tsx [hunk 1]
```diff
@@ -39,19 +39,18 @@ export const TopicList = React.memo(
...
 
-    const { data, loading, error } = useQuery<TData>(GET_ROLES, {
+    const { data, loading, error } = useQuery<TData>(GET_TOPICS, {
       variables: {
-        id: resourceId,
+        where: { 
+          resource: { id: resourceId }, 
+          displayName: {
+            contains: searchPhrase,
+            mode: models.QueryMode.Insensitive,
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Topics/TopicList.tsx:48]: Did you test what happens when searchPhrase is an empty string?

File: packages/amplication-server/src/core/build/build.service.ts [hunk 2]
```diff
@@ -597,7 +597,9 @@ export class BuildService {
     rootGeneration = true
   ): Promise<CodeGenTypes.DSGResourceData> {
     const resources = await this.resourceService.resources({
-      where: { project: { resources: { some: { id: resourceId } } } }
+      where: {
+        project: { resources: { some: { id: { equals: resourceId } } } }
+      }
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/build/build.service.ts:602]: Is that just a refactor or is it a bug fix? 
I think @tupe12334 is working on this exact code right now and it looks like a potential for an unnecessary conflict

File: packages/amplication-server/src/core/resource/dto/ResourceWhereInput.ts [hunk 3]
```diff
@@ -7,10 +7,10 @@ import { EnumResourceTypeFilter } from './EnumResourceTypeFilter';
   isAbstract: true
 })
 export class ResourceWhereInput {
-  @Field(() => String, {
+  @Field(() => StringFilter, {
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/resource/dto/ResourceWhereInput.ts:10]: This should be 
StringFilter | String

File: packages/amplication-server/src/core/resource/resource.service.spec.ts [hunk 4]
```diff
@@ -592,7 +592,7 @@ describe('ResourceService', () => {
     const args = {
       where: {
         deletedAt: null,
-        id: EXAMPLE_RESOURCE_ID
+        id: { equals: EXAMPLE_RESOURCE_ID }
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/resource/resource.service.spec.ts:595]: see my comments on ResourceWhereInput.ts

File: packages/amplication-server/src/core/resource/resource.service.ts [hunk 5]
```diff
@@ -317,6 +319,22 @@ export class ResourceService {
     });
   }
 
+  async messageBrokerConnectedServices(args: FindOneArgs): Promise<Resource[]> {
+    const resource = await this.resource(args);
+    const serviceTopicsCollection = await this.serviceTopicsService.findMany(
+      {}
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/resource/resource.service.ts:325]: filter serviceTopics by relation to the current project only - you are now using all serviceTopic in the system from any project and any workspace

File: packages/amplication-server/src/core/serviceTopics/dto/ServiceTopicsWhereInput.ts [hunk 6]
```diff
@@ -1,7 +1,13 @@
...
 
 @InputType({
   isAbstract: true
 })
-export class ServiceTopicsWhereInput extends BlockTypeWhereInput {}
+export class ServiceTopicsWhereInput extends BlockTypeWhereInput {
+  @Field(() => StringFilter, {
+    nullable: true
+  })
+  messageBrokerId?: StringFilter | null;
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/serviceTopics/dto/ServiceTopicsWhereInput.ts:12]: are you using this?

File: packages/amplication-server/src/core/serviceTopics/serviceTopics.service.ts [hunk 7]
```diff
@@ -43,7 +44,7 @@ export class ServiceTopicsService extends BlockTypeService<
         project: {
           id: resource.projectId
         },
-        id: messageBrokerId,
+        id: { equals: messageBrokerId },
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/serviceTopics/serviceTopics.service.ts:47]: see my comments above and revert this code please

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-server/src/core/serviceTopics/dto/ServiceTopicsWhereInput.ts (hunk 6 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/serviceTopics/dto/ServiceTopicsWhereInput.ts:12]: Nope. Removed.

File: packages/amplication-server/src/core/build/build.service.ts (hunk 2 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/build/build.service.ts:602]: Done.

File: packages/amplication-server/src/core/resource/resource.service.ts (hunk 5 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/resource/resource.service.ts:325]: Done

File: packages/amplication-server/src/core/serviceTopics/serviceTopics.service.ts (hunk 7 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/serviceTopics/serviceTopics.service.ts:47]: Done

File: packages/amplication-client/src/Topics/TopicList.tsx (hunk 1 above)
Comment (EugeneTseitlin) [reply packages/amplication-client/src/Topics/TopicList.tsx:48]: Reverted change.

File: packages/amplication-server/src/core/resource/dto/ResourceWhereInput.ts (hunk 3 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/resource/dto/ResourceWhereInput.ts:10]: Not supported.

File: packages/amplication-server/src/core/resource/resource.service.spec.ts (hunk 4 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/resource/resource.service.spec.ts:600]: Done

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**